pinot-plugins/pinot-stream-ingestion/pinot-kinesis/src/main/java/org/apache/pinot/plugin/stream/kinesis/KinesisMessageBatch.java

```diff
@@ -31,12 +31,22 @@ public class KinesisMessageBatch implements MessageBatch<byte[]> {
   private final List<BytesStreamMessage> _messages;
   private final KinesisPartitionGroupOffset _offsetOfNextBatch;
   private final boolean _endOfShard;
+  private final boolean _isMissingOffsets;
 
   public KinesisMessageBatch(List<BytesStreamMessage> messages, KinesisPartitionGroupOffset offsetOfNextBatch,
       boolean endOfShard) {
     _messages = messages;
     _offsetOfNextBatch = offsetOfNextBatch;
     _endOfShard = endOfShard;
+    _isMissingOffsets = false;
+  }
+
+  public KinesisMessageBatch(List<BytesStreamMessage> messages, KinesisPartitionGroupOffset offsetOfNextBatch,
```

Comment:
No need to add this new constructor. We can add it when data loss can be detected